libsolidity/parsing/Token.h

```diff
@@ -312,7 +312,12 @@ class Token
 	static bool isVisibilitySpecifier(Value op) { return isVariableVisibilitySpecifier(op) || op == External; }
 	static bool isVariableVisibilitySpecifier(Value op) { return op == Public || op == Private || op == Internal; }
 	static bool isLocationSpecifier(Value op) { return op == Memory || op == Storage || op == CallData; }
-	static bool isStateMutabilitySpecifier(Value op) { return op == Pure || op == Constant || op == View || op == Payable; }
+	static bool isStateMutabilitySpecifier(Value op, bool _allowConstant = true)
```

Comment:
Shouldn't we just remove `constant` from here for 0.5.0? Or do we need it to output recommendations?

Reply:
So far we still have error messages indicating that ``constant`` is replaced by ``view`` for state mutability - that's why it's still in here. We have to exclude it for ``address``, so that we can have ``address constant`` and ``address payable constant``, though. We could go ahead and remove it entirely for 0.5.0, but I think it may make sense to keep the detailed suggestions for now.